asabiris/output/smtp/service.py

```
@@ -277,6 +345,134 @@ async def send(
+			encoded = base64.b64encode(auth_string.encode("utf-8")).decode("ascii")
+			headers.append("Proxy-Authorization: Basic {}".format(encoded))
+		return "\r\n".join(headers) + "\r\n\r\n"
+
+	async def _read_http_headers(self, sock_obj):
+		loop = asyncio.get_running_loop()
+		buffer = b""
+		max_headers_size = 65536
+		while b"\r\n\r\n" not in buffer:
+			chunk = await loop.sock_recv(sock_obj, 4096)
+			if not chunk:
+				raise ProxyConnectError("Proxy closed connection before CONNECT response")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)